base/log.hpp

```diff
@@ -72,4 +55,57 @@ void log_msg(const std::string& msg);
 // TODO(legend): may port more functions here from glog.
 
 }  // namespace base
+
+/// Because of including the gperftools/malloc_extension.h of TCMALLOC,
+/// there is a conflict between the namespace ::base which is belonging to gperftools
+/// and the namaspace husky::base. In order to slove this problem, we use 2 warpper classes
+/// "HuskyLogger_Warpper" and "HuskyLoggerVoidify_Warpper" in namespace husky.
```

Comment:
It depends on the compiler version. If we use VS2015 or G++ 6.1, it will be ok. However, it will report  an error when using g++ 5.2